pkg/controller/garbagecollector/garbagecollector.go

```diff
@@ -472,7 +472,7 @@ func (gc *GarbageCollector) classifyReferences(ctx context.Context, item *node,
 		if err != nil {
 			return nil, nil, nil, err
 		}
-		if ownerAccessor.GetDeletionTimestamp() != nil && hasDeleteDependentsFinalizer(ownerAccessor) {
+		if ownerAccessor.GetDeletionTimestamp() != nil && hasDeleteDependentsFinalizer(ownerAccessor) && len(ownerAccessor.GetFinalizers()) == 1 {
```

Comment:
I am a little confused as to why we need to add this condition.

Reply:
This is for ensuring the owner object only has `foregroundDeletion` finalizer.
(= other related finalizers are done, which should have same consistency as when GC is triggered with Background)